dxr/plugins/python/tests/test_parse_errors.py

```diff
@@ -0,0 +1,70 @@
+import warnings
+from textwrap import dedent
+
+from dxr.plugins.python.tests import PythonSingleFileTestCase
+
+
+class WarningTestCase(PythonSingleFileTestCase):
+    @classmethod
+    def get_config(cls, config_dir_path):
+        # We need to disable workers because catch_warnings does not
+        # support multi-threaded or multi-process capturing.
+        config = super(WarningTestCase, cls).get_config(config_dir_path)
+        config['DXR']['workers'] = 0
+        return config
+
+    @classmethod
+    def setup_class(cls):
+        # Wrap the normal setup to catch warnings.
+        with warnings.catch_warnings(record=True) as caught_warnings:
+            warnings.simplefilter('always')
+            super(WarningTestCase, cls).setup_class()
+            cls.warnings = caught_warnings
+
+
+class InvalidPythonTest(WarningTestCase):
+    source = dedent("""
+    from foo import bar
```

Comment:
Let's indent these strings one more notch so other class vars, if they existed, would stick out. That's the convention we were using.